### DISCUSSION TITLE
经过bfe，response body数据丢失的问题

### DISCUSSION THREAD

**Discussion by lf2186** [bfenetworks/bfe#1033] [Q&A (问题和回答)]
hi，

    大佬们好，由于ssl的原因，我们临时架构是用户->nginx->bfe->api网关->服务；

现在碰到问题，nginx直接到api网关，业务正常，但是中间有bfe以后，在nginx上抓包，看bfe的response body为空了。

a）在nginx抓包为：
<img width="582" alt="image" src="https://user-images.githubusercontent.com/4362152/159214090-9d4e3c72-3839-4330-a2a8-29f01e6c867f.png">

b）在网关上抓包：
<img width="659" alt="image" src="https://user-images.githubusercontent.com/4362152/159214195-c8d553b5-6aa3-4928-97b4-96018f54f0d1.png">

对比了一下有没有bfe的差别，经过bfe以后，request http version改为1.1，response 多了transfer-encoding:chunked
请问是什么原因导致，增加bfe以后，body丢失的？

**Comment by daimg** [comment]
你设置的resFlushInterval参数是什么值？
因为你的服务/api网关返回了chunked格式的response，如果设置了resFlushInterval=0，bfe将有可能需要等待所有的chunk，才能返回response给client。
